### PR TITLE
[Issue #34] Basic Chat UI

### PR DESCRIPTION
**Pull request by mohameddahrouj**
**Overview**
- Access at **#/chats** 
- Hooked up to mock data
- Use of directives for separating user chats and current conversation.
- Added a few packages (Added sweetalert too for the future...)

### CODE REVIEW

File: SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    display: none;
+}
+.chat .chat-wrapper .chat-list-wrapper .chat-list li.active {
+    color: black;
+}
+.chat .chat-wrapper .chat-list-wrapper .chat-list li.active .body .timestamp {
+    color: black;
+}
+.chat .chat-wrapper .chat-list-wrapper .chat-list li.new {
+
```
Comment (lavatahir) [inline SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.css]: If empty do we need it?

File: SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            $(".chat-list-wrapper").niceScroll({autohidemode:'leave'});
+            $scope.setActive = function(contact) {
+                if($scope.activeContact !== contact) {
+                    $scope.activeContact = contact;
+                    $scope.contactSelected({'contact' : contact});
+                }
+            };
+
+           $scope.$watch('contacts', function(contacts) {
+              if (contacts !== undefined && contacts.length > 0) {
```
Comment (lavatahir) [inline SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js]: Might need to check if null if i remember js has undefined and null being different. You Can just use truthy falsey instead

File: SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.css (hunk 1 above)
Comment (mohameddahrouj) [reply SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.css]: good call

File: SECP-service/src/main/resources/assets/app/index.html [hunk 3]
```diff
@@ -1,65 +1,68 @@
...
+                        <span class="icon-bar"></span>
+                        <span class="icon-bar"></span>
+                    </button>
+                    <a class="navbar-brand" href="/#/">SECP</a>
+                </div>
+                <!-- Collect the nav links, forms, and other content for toggling -->
+                <div class="collapse navbar-collapse navbar-ex1-collapse">
+                    <ul class="nav navbar-nav">
+                        <li>
+                            <a href="/#/login">Login</a>
```
Comment (tosinoni) [inline SECP-service/src/main/resources/assets/app/index.html]: I changed this in my latest PR. I got rid of the hash-tags.

File: SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        };
+
+      $scope.contactSelected = function(contact) {
+         $scope.selectedChat = contact;
+         Chat.getMessages(contact).then(function(data) {
+            $scope.messages = data;
+         });
+      };
+
+      $scope.create = function () {
```
Comment (tosinoni) [inline SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js]: we don't need all the methods below contactSelected. I just forgot to delete them.

File: SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js (hunk 2 above)
Comment (mohameddahrouj) [reply SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js]: @tosinoni this is your intended behaviour right?

File: SECP-service/src/main/resources/assets/app/index.html (hunk 3 above)
Comment (mohameddahrouj) [reply SECP-service/src/main/resources/assets/app/index.html]: I will change it now

File: SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js (hunk 4 above)
Comment (mohameddahrouj) [reply SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js]: Ill get rid of them

**Comment by mohameddahrouj** [comment]
Good to go, only outstanding comment is Lava's comment on chat-list.js regarding the null check

File: SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js (hunk 2 above)
Comment (tosinoni) [reply SECP-service/src/main/resources/assets/app/directives/chat-list/chat-list.js]: It is the intended behaviour. 

A property that has no definition is undefined, which also means no value, no type and never been referenced before in the scope. On the other hand, null is where the property exists but it hasn't known what its value is.

**Pull request merged**